hummingbot/core/utils/market_mid_price.py

```diff
@@ -0,0 +1,96 @@
+import requests
+from decimal import Decimal
+from typing import Optional
+from hummingbot.market.binance.binance_market import BinanceMarket
+from hummingbot.market.kraken.kraken_market import KrakenMarket
+
+
+BINANCE_PRICE_URL = "https://api.binance.com/api/v3/ticker/bookTicker"
+KUCOIN_PRICE_URL = "https://api.kucoin.com/api/v1/market/allTickers"
+LIQUID_PRICE_URL = "https://api.liquid.com/products"
+BITTREX_PRICE_URL = "https://api.bittrex.com/api/v1.1/public/getmarketsummaries"
+KRAKEN_PRICE_URL = "https://api.kraken.com/0/public/Ticker?pair="
+COINBASE_PRO_PRICE_URL = "https://api.pro.coinbase.com/products/TO_BE_REPLACED/ticker"
+
+
+def get_mid_price(exchange: str, trading_pair: str) -> Optional[Decimal]:
+    if exchange == "binance":
+        return binance_mid_price(trading_pair)
+    elif exchange == "kucoin":
+        return kucoin_mid_price(trading_pair)
+    elif exchange == "liquid":
+        return liquid_mid_price(trading_pair)
+    elif exchange == "bittrex":
+        return bittrex_mid_price(trading_pair)
+    elif exchange == "kraken":
+        return kraken_mid_price(trading_pair)
+    elif exchange == "coinbase_pro":
+        return coinbase_pro_mid_price(trading_pair)
+    else:
+        return binance_mid_price(trading_pair)
+
+
+def binance_mid_price(trading_pair: str) -> Optional[Decimal]:
```

Comment:
Please add `@cachetools.func.ttl_cache(...)` decorators to all the fetching functions here s.t. if some strategy code calls them repeatedly, it won't produce excessive delays in the UI.